pyccel/codegen/printing/fcode.py

```diff
@@ -719,15 +715,16 @@ def _print_NumpyFloor(self, expr):
     # ======================================================================= #
     def _print_PyccelArraySize(self, expr):
         init_value = self._print(expr.arg)
-
+        prec = iso_c_binding["integer"][expr.precision]
         if expr.arg.order == 'C':
             index = self._print(expr.arg.rank - expr.index)
         else:
             index = self._print(expr.index + 1)
 
-        code_init = 'size({0}, {1})'.format(init_value, index)
+        if expr.arg.rank == 1:
+            return 'size({0}, kind={1})'.format(init_value, prec)
 
-        return code_init
+        return 'size({0}, {1}, {2})'.format(init_value, index, prec)
```

Comment:
Same comment as above: in order to have line 727 consistent with line 725, shouldn't we pass `prec` through the `kind` keyword argument?

Reply:
same as above